dependency/dependency.go

```diff
@@ -438,6 +439,11 @@ func (c *Client) checkUpstreamVersions(deps []*Dependency) ([]versionUpdateInfo,
 	versionUpdates := []versionUpdateInfo{}
 	for _, dep := range deps {
 		if dep.Upstream == nil {
+			versionUpdates = append(versionUpdates, versionUpdateInfo{
+				name:            dep.Name,
+				current:         Version{dep.Version, dep.Scheme},
+				updateAvailable: false,
+			})
```

Comment:
I didn't think of entry-without-upstream as a valid case for dependency tracking, but it makes total sense to support it 👍

Reply:
@Pluies  It was supported by the existing code, I guess it is to keep a version number in sync in multiple files?

The reason I discovered it was that that the first thing I tried was to upgrade `testdata/local.yaml`, and the result was that `local.yaml` only containted `dependencies: []` afterwards. That was very confusing :D